src/util-misc.c

```diff
@@ -933,235 +645,136 @@ static int UtilMiscParseSizeStringTest01(void)
+    FAIL_IF(result != 10.5 * 1024);
 
     str = "10.5 Kb ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024);
 
     str = "10.5 KB ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024);
 
     str = "10.5 mb ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024 * 1024);
 
     str = "10.5 gb ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024 * 1024 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024 * 1024 * 1024);
 
     /* space start - space end */
 
     str = " 10.5 ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5);
 
     str = " 10.5 kb ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024);
 
     str = " 10.5 Kb ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024);
 
     str = " 10.5 KB ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024);
 
     str = " 10.5 mb ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024 * 1024);
 
     str = " 10.5 gb ";
     result = 0;
-    if (ParseSizeString(str, &result) > 0) {
-        goto error;
-    }
-    if (result != 10.5 * 1024 * 1024 * 1024) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024 * 1024 * 1024);
 
     /* Should fail on unknown units. */
-    if (ParseSizeString("32eb", &result) > 0) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString("32eb", &result) > 0);
 
     return 1;
```

Comment:
This should be made `PASS`

Reply:
understood, I'll make the change and create a new PR. Thanks!

Reply:
@inashivb should I let the name of the new PR be the same as this one? And should I add the link of this PR in the description of the new one?

Reply:
> @inashivb should I let the name of the new PR be the same as this one?

Yes, please, create a new PR with the same title as this one, adding a version to it (so: `util-misc: Convert unittests to new FAIL/PASS API - v2`)

> And should I add the link of this PR in the description of the new one?

Yes, and then you can close this one with a note that work is continued on the new PR, and add a link :)

Reply:
will do, thanks!